db.js

```diff
@@ -0,0 +1,51 @@
+const knex = require('knex');
+const knexConfig = require('./knexfile.js');
+const db = knex(knexConfig.development);
+
+module.exports = {
+    getDishes,
+    addDish,
+    getDish,
+    getRecipes,
+    addRecipe,
+    getRecipe,
+    getShoppingList
+};
+
+function getDishes(){
+    return db('dishes')
+}
+
+function getDish(id){
+    return db('dishes').where({ id: Number(id) });
+}
+
+function addDish(dish) {
+    return db('dishes')
+    .insert(dish)
+    .then(ids => ({ id: ids[0] }));
+}
+
+function getRecipes(){
+    return db('recipes')
+}
+
+function getRecipe(recipe){
+    return db('recipes')
+    .insert(recipe)
+    .then(ids => ({ id: ids[0] }));
+}
+
+function addRecipe(recipe){
+    return db('recipes')
+    .insert(recipe)
+    .then(ids => ({ id: ids[0] }));
+}
+
+function getShoppingList(id){
+    return db('recipes as r').where('r.id', id)
+    .join('dishes as d', 'r.dish_id', '=', 'd.id')
```

Comment:
Good job with the joins here. It's important to think about how you want your site to function. We're just getting an ingredient list by recipe_id. We probably want this to be returned any time a query is made for recipe by id and the object should be shaped something like this: 
```
{
    id: 1,
    recipe_name: "pork street taco",
    dish_id: 4,
    dish_name: "taco",
    ingredients: [
        {
            ingredient_id: 12,
            ingredient: "tortilla",
            quantity: 4
        },
        {other ingredients....},
        {other ingredients....}
    ]
}

```